src/black/files.py

```diff
@@ -211,15 +211,27 @@ def gen_python_files(
 
     `report` is where output about exclusions goes.
     """
+
+    def is_ignored(
+        gitignore_dict: Dict[Path, PathSpec], child: Path, report: Report
+    ) -> bool:
+        for _dir, _gitignore in gitignore_dict.items():
+            relative_path = normalize_path_maybe_ignore(child, _dir, report)
+            if relative_path is None:
+                continue
```

Comment:
Seems like `return False` is a better alternative since a symlink that goes back inside root should be ignored (at least seems to be ignored now)

Reply:
Yeah, seems like the `continue` produces additional unnecessary output. I'll replace it with `break` so that the function returns `False`